packages/app/src/cli/services/dev/extension/bundler.ts

```diff
@@ -24,11 +24,19 @@ export async function setupBundlerAndFileWatcher(options: FileWatcherOptions) {
   const bundlers: Promise<void>[] = []
 
   options.devOptions.extensions.forEach((extension) => {
+    const stdinContent = extension.entrySourceFilePaths
```

Comment:
I think we might have a problem making this work with the automatically calling `shopify.extend` for UI extension as we won't know when we can simply call `import './${path.relative(extension.directory, filePath)}';` vs. needing to do 
```ts
import * as ExtensionModule1 from './${path.relative(extension.directory, filePath)}';`;

if ('default' in ExtensionModule1) {
  // Inject the code to register the extension point
  shopify.extend(
    extension.target, 
    ExtensionModule1,
} else {
  // With no default export, we assume they used the global API directly
}
```

Reply:
I think we just a have a conditional here to generate the stinConten only if it's a `ui_extension`

```ts
options.devOptions.extensions.forEach((extension) => {
    const stdin = extension.type === 'ui_extension' ? getUIExtensionIndex(extension) : undefined;

    bundlers.push(
      bundleExtension({
        minify: false,
        outputBundlePath: extension.outputBundlePath,
        sourceFilePaths: stdinContent ? undefined : extension.entrySourceFilePath,,
        stdin,
        ...
      })
   )
 });
```

Reply:
In a previous version I did something similar to this, but I thought it better to avoid the conditional logic since all extension points can be built using stdin.

But, I think you're right.  Will revert.